src/main/java/io/nats/client/impl/NatsConnectionWriter.java

```diff
@@ -193,11 +207,11 @@ public void run() {
 
             while (running.get() && !Thread.interrupted()) {
                 NatsMessage msg;
-                if (this.reconnectMode.get()) {
-                    msg = this.reconnectOutgoing.accumulate(sendBufferLength.get(), Options.MAX_MESSAGES_IN_NETWORK_BUFFER, reconnectTimeout);
+                if (mode.get() == Mode.Normal) {
+                    msg = this.normalOutgoing.accumulate(sendBufferLength.get(), Options.MAX_MESSAGES_IN_NETWORK_BUFFER, outgoingTimeout);
                 }
                 else {
-                    msg = this.outgoing.accumulate(sendBufferLength.get(), Options.MAX_MESSAGES_IN_NETWORK_BUFFER, outgoingTimeout);
+                    msg = this.reconnectOutgoing.accumulate(sendBufferLength.get(), Options.MAX_MESSAGES_IN_NETWORK_BUFFER, reconnectTimeout);

```

Comment:
not normal mode is both the others, we need to get all the messages from the reconnect queue until the end mark is found, and then we can switch to the normal mode.